pkg/sql/ttl/ttljob/ttljob_test.go

```diff
@@ -299,17 +345,98 @@ INSERT INTO t (id, crdb_internal_expiration) VALUES (1, now() - '1 month'), (2,
 			th.sqlDB.ExecMultiple(t, strings.Split(tc.setup, ";")...)
 
 			// Force the schedule to execute.
-			th.env.SetTime(timeutil.Now().Add(time.Hour * 24))
-			require.NoError(t, th.executeSchedules())
-
 			th.waitForScheduledJob(t, jobs.StatusFailed, tc.expectedTTLError)
+
 			var numRows int
 			th.sqlDB.QueryRow(t, `SELECT count(1) FROM t`).Scan(&numRows)
 			require.Equal(t, 2, numRows)
 		})
 	}
 }
 
+func TestRowLevelTTLJobMultipleNodes(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	numNodes := 5
+	th, cleanupFunc := newRowLevelTTLTestJobTestHelper(
+		t,
+		&sql.TTLTestingKnobs{
+			AOSTDuration:             &zeroDuration,
+			ReturnStatsError:         true,
+			ExpectedNumSpanParitions: 2,
+		},
+		false,    /* testMultiTenant */ // SPLIT AT does not work with multi-tenant
+		numNodes, /* numNodes */
+	)
+	defer cleanupFunc()
+
+	sqlDB := th.sqlDB
+
+	// create table
+	tableName := "tbl"
+	sqlDB.Exec(t, fmt.Sprintf(
+		`CREATE TABLE %s (
+			id INT PRIMARY KEY,
+			expire_at TIMESTAMP
+			) WITH (ttl_expiration_expression = 'expire_at')`,
+		tableName,
+	))
+
+	// split table
+	splitAt := 10_000
```

Comment:
should we make `splitAt` and `rowsPerRange` const?

Reply:
Done